client/src/pages/proverb/Proverb.js

```diff
@@ -4,14 +4,24 @@ import Input from "./../../components/Input";
 import Button from "./../../components/Button";
 import { isArabic } from "./../../helpers/functions";
 import { PROVERB_INITIAL_DATA } from "./../../helpers/formData";
-import { useProverb } from "./../../redux/hooks";
+import { useProverb, useAuth } from "./../../redux/hooks";
 
 import "./Proverb.css";
 
 const Proverb = ({ actionType, handleCloseModal }) => {
-  const { addProverb } = useProverb();
+  const { isAuthenticated } = useAuth();
+  const {
+    addProverb,
+    addUserProverb,
+    updateProverb,
+    proverb: proverbObj,
+    loading,
+  } = useProverb();
 
-  const [formData, setFormData] = useState(PROVERB_INITIAL_DATA);
+  const isNewProverb = !loading && actionType === "Add";
+  const proverbFormValues = isNewProverb ? PROVERB_INITIAL_DATA : proverbObj;
```

Comment:
awesome :) much more readable!

Reply:
Thanks,it was @frankPairs  solution.  :))